bundles/org.openhab.core/src/main/java/org/openhab/core/service/AbstractWatchService.java

```diff
@@ -37,14 +37,25 @@
 public abstract class AbstractWatchService {
     protected @Nullable String pathToWatch;
 
+    /**
+     * The queue reader
+     */
+    protected WatchQueueReader watchQueueReader = WatchQueueReader.getInstance();
+
     protected AbstractWatchService(String pathToWatch) {
         this.pathToWatch = pathToWatch;
     }
 
     /**
-     * The queue reader
+     * Change the watch directory for this WatchService
+     *
+     * @param pathToWatch the new path
      */
-    protected WatchQueueReader watchQueueReader = WatchQueueReader.getInstance();
+    protected void changeWatchDirectory(String pathToWatch) {
+        watchQueueReader.stopWatchService(this);
```

Comment:
Would it also possible to call `deactivate()` here instead? Subclasses would then have a way to cleanup stuff that gets added/created in `activate()`.